### PR TITLE
Add `// NOTCONSOLE` to docs

### PR DESCRIPTION
**Pull request by nik9000**
We have 1074 snippets that look like they should be converted to
`// CONSOLE`. At least that is what `gradle docs:listConsoleCandidates`
says. This adds `// NOTCONSOLE` to explicitly mark snippets that
_shouldn't_ be converted to `// CONSOLE`. After marking the blindingly
obvious ones this cuts the remaining snippet count to 1032.

Relates to #18160

### CODE REVIEW

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/doc/SnippetsTask.groovy [hunk 1]
```diff
@@ -114,23 +114,38 @@ public class SnippetsTask extends DefaultTask {
                     return
                 }
                 if (line ==~ /\/\/\s*AUTOSENSE\s*/) {
-                    throw new InvalidUserDataException("AUTOSENSE has been " +
-                            "replaced by CONSOLE. Use that instead at " +
-                            "$file:$lineNumber")
+                    throw new InvalidUserDataException("$file:$lineNumber: "
```
Comment (nik9000) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/doc/SnippetsTask.groovy]: I took the opportunity normalize the error reporting here.

**Comment by nik9000** [comment]
@clintongormley, can you have a look? I'd like to add something like this so I know which candidates are safe to skip when I list them.

I think it'd be nice one day to fail the build for snippets that aren't explicitly `// CONSOLE` or `// NOTCONSOLE`. Meaning you *have* to make a decision about the snippet when you write it.

**Comment by clintongormley** [comment]
LGTM

**Comment by clintongormley** [comment]
Actually, couldn't we only require the use of `NOTCONSOLE` for source blocks which use `js` (or `javascript` or `json`)?

**Comment by nik9000** [comment]
I thought about that. We often use `sh` and `shell` for `curl` examples which I think we should be thinking about converting as well so I think we should add it to those.

I'm not actually requiring it on anything right yet, though I might open something for that soon.

I already ban `json` because the syntax highlighter only know `js`. We only have one `javascript`. I should probably ban that too for the same reason but I'm not sure. What do you think?

**Comment by clintongormley** [comment]
> We only have one javascript. I should probably ban that too for the same reason but I'm not sure. What do you think?
agreed

**Pull request merged**